internal/util/get/get.go

```diff
@@ -84,18 +86,30 @@ func (c Command) Run(ctx context.Context) error {
 		UpdateStrategy: c.UpdateStrategy,
 	}
 
-	err = kptfileutil.WriteFile(c.Destination, kf)
+	err = kptfileutil.WriteFile(tempPath, kf)
 	if err != nil {
-		return errors.E(op, types.UniquePath(c.Destination), err)
+		return errors.E(op, types.UniquePath(tempPath), err)
 	}
 
-	p, err := pkg.New(c.Destination)
+	p, err := pkg.New(tempPath)
 	if err != nil {
-		return errors.E(op, types.UniquePath(c.Destination), err)
+		return errors.E(op, types.UniquePath(tempPath), err)
 	}
+	p.DisplayPath = types.DisplayPath(filepath.Base(c.Destination))
 
 	if err = c.fetchPackages(ctx, p); err != nil {
-		return errors.E(op, types.UniquePath(c.Destination), err)
+		return errors.E(op, types.UniquePath(tempPath), err)
+	}
+
+	// Move temporary directory into final destination.
+	err = os.MkdirAll(filepath.Dir(c.Destination), 0700)
```

Comment:
Have you explored clean-up on error option as opposed to writing to temp and copying over? If get behavior is dependent on parent package state then this approach will not work. Previously, setter inheritance used to depend on package structure so this approach wouldn't work. But I feel it is better to implement clean up on failure. There might be features in the future where get might depend on parent packages.

@mortent

Reply:
I also think cleaning up on failure would be a better option here, unless there is a reason why that wouldn't work.

Reply:
Done.